tests/unit/ml_handlers/test_timegpt.py

```diff
@@ -0,0 +1,102 @@
+            if not ret.empty:
+                if ret["STATUS"][0] == "complete":
+                    done = True
+                    break
+                elif ret["STATUS"][0] == "error":
+                    raise RuntimeError("predictor failed", ret["ERROR"][0])
+            time.sleep(0.5)
+        if not done:
+            raise RuntimeError("predictor wasn't created")
+
+    def run_sql(self, sql):
+        ret = self.command_executor.execute_command(parse_sql(sql, dialect="mindsdb"))
+        assert ret.error_code is None
+        if ret.data is not None:
+            columns = [col.alias if col.alias is not None else col.name for col in ret.columns]
+            return pd.DataFrame(ret.data, columns=columns)
+
+    def test_missing_required_keys(self):
+        # create project
+        self.run_sql("create database proj")
+        self.run_sql(f"""create ml_engine timegpt from timegpt using api_key='{self.get_api_key()}';""")
+        # with pytest.raises(Exception):
+        self.run_sql(
+            """
+              create model proj.test_timegpt_missing_required_keys
+              predict answer
+              using
+                engine='timegpt';
+           """
+        )
+
+    def test_unknown_arguments(self):
+        self.run_sql("create database proj")
+        self.run_sql(f"""create ml_engine timegpt from timegpt using api_key='{self.get_api_key()}';""")
+        with pytest.raises(Exception):
+            self.run_sql(
+                f"""
+                create model proj.test_timegpt_unknown_arguments
+                predict answer
+                using
+                    engine='timegpt',
+                    api_key='{self.get_api_key()}',
+                    evidently_wrong_argument='wrong value';  --- this is a wrong argument name
+            """
+            )
+
+    @patch("mindsdb.integrations.handlers.postgres_handler.Handler")
+    def test_forecast_group(self, mock_handler):
+        # create project
+        self.run_sql("create database proj")
+        df = pd.read_csv('tests/unit/ml_handlers/data/house_sales.csv')
+        self.set_handler(mock_handler, name="pg", tables={"df": df})
+
+        self.run_sql(f"""create ml_engine timegpt from timegpt using api_key='{self.get_api_key()}';""")
+
+        self.run_sql(
+            f"""
+           create model proj.test_timegpt_forecast
+           predict ma
+           order by saledate
+           group by type, bedrooms
+           window 128
+           horizon 5
+           using
+             engine='timegpt',
+             api_key='{self.get_api_key()}';
+        """
+        )
+        self.wait_predictor("proj", "test_timegpt_forecast")
+
+        self.run_sql(
+            """
+            SELECT p.ma
+            FROM proj.test_timegpt_forecast as p
+            JOIN pg.df as t
+            WHERE p.saledate > LATEST;
+        """
+        )
+
+        # asserts
```

Comment:
Looks great. Are we missing asserts here ?

Reply:
Absolutely, that's a great catch! I will add some as soon as the API key gets recognized by the CI

Reply:
Added :white_check_mark: